docs/index.md

```diff
@@ -1,94 +1,217 @@
+    const int32_t height = static_cast<int32_t>(videoMode->height * 0.9f);
+
+    glfwWindowHint(GLFW_SCALE_TO_MONITOR, GLFW_FALSE);
+    glfwWindowHint(GLFW_CLIENT_API, GLFW_NO_API);
+    const GLFWwindow* window = glfwCreateWindow(
+        width,
+        height,
+        "LearnD3D11 - Hello Window",
+        nullptr,
+        nullptr);
+    if (window == nullptr)
+    {
+        std::cout << "GLFW: Unable to create window\n";
+        glfwTerminate();
+        return -1;
+    }
+
+    const int32_t windowLeft = videoMode->width / 2 - width / 2;
+    const int32_t windowTop = videoMode->height / 2 - height / 2;
+    glfwSetWindowPos(window, windowLeft, windowTop);
+
+    while (!glfwWindowShouldClose(window))
     {
-        public SomeCSharpCodeHere()
-        {
-        }
+        glfwPollEvents();
+        // future update code
+        // future render code
     }
-    ```
 
-    ```xml
-    <Why Not="Some">
-        <Xml/>
-    </Why>
-    ```
+    glfwDestroyWindow(window);
+    glfwTerminate();
+    return 0;
+}
+```
+
+Now let me explain what all these things mean.
+
+```cpp
+#include <GLFW/glfw3.h>
+```
+
+C++ needs to know where all the definitions and declarations are coming from.
+
+```cpp
+    if (!glfwInit())
+    {
+        std::cout << "GLFW: Unable to initialize\n";
+        return -1;
+    }
+```
+
+Pretty obvious, right? Try to initialize `GLFW`. If it fails to do so, let the user know and end the program, since there is no point in going further.
+
+```cpp
+    const GLFWmonitor* primaryMonitor = glfwGetPrimaryMonitor();
+    const GLFWvidMode* videoMode = glfwGetVideoMode(primaryMonitor);
+    const int32_t width = static_cast<int32_t>(videoMode->width * 0.9f);
+    const int32_t height = static_cast<int32_t>(videoMode->height * 0.9f);
+```
+
+This piece of code grabs the main monitor's resolution, so that we can derive a window width and height from it - and it will look similar no matter what resolution you use.
+
+```cpp
+    glfwWindowHint(GLFW_SCALE_TO_MONITOR, GLFW_FALSE);
+```
+
+This will tell `GLFW` to not scale the window in any way, should you have setup a specific scaling other than 100% on your desktop. That will keep the windowsize at what we set it, without thinking about odd fractionals to manually scale the windowsize for any arbitrary scaling on your OS.
+
+```cpp
+    glfwWindowHint(GLFW_CLIENT_API, GLFW_NO_API);
+```
+
+`GLFW` was initially meant to support development of OpenGL based applications, hence the gl in its name, but over the years it also supports other APIs not just OpenGL. Now since `GLFW` automatically create a context for OpenGL we can tell it not to do it via that window hint.
+
+```cpp
+    const int32_t windowLeft = videoMode->width / 2 - width / 2;
+    const int32_t windowTop = videoMode->height / 2 - height / 2;
+    glfwSetWindowPos(window, windowLeft, windowTop);
+```
+
+Should be fairly obvious. We like centered windows.
```

Comment:
I recommend to explain what glfwSetWindowPos does a bit more... why does it center a window?

"Should be fairly obvious. We like centered windows — and glfwSetWindowPos allow us to specify a window to set location to.
in this case — `window` is the GLFW window we created. Since top left of a window is the position of the window we set the window top left position by getting half the monitor size subtracted by the window we created size, getting us windowLeft and windowTop respectivly"

Reply:
Do we really have to dumb it down that much?

Reply:
true, but else we ask the reader to search through the code to find what "videoMode" is; plus top left of window being position is not intuitive unless you previously worked with GUI windowing before.

plus we also say things overly simplified such as: "Now we clean up any resource we created, such as the window itself and the GLFW system. Then simply return to the OS, without any error."